lib/stores/S3Store.js

```diff
@@ -505,14 +505,30 @@ class S3Store extends DataStore {
             this._getMetadata(file_id)
                 .then((metadata) => {
                     return this._retrieveParts(file_id)
-                        .then((parts) => {
-                            return {
-                                parts,
-                                metadata,
-                            };
-                        })
+                        .then((parts) => ({ parts, metadata }))
                         .catch((err) => {
-                            throw err;
+                            // getOffset is called from both the HEAD request and the PATCH request
+                            // (via the _processUpload method). The HEAD request calls it (implicitly)
+                            // with with_parts: false, the _processUpload method calls it (explicitly)
+                            // with with_parts: true. The _processUpload method expects upload parts to be returned,
+                            // which would not work when the upload is completed, since there are no parts anymore.
+                            // TODO: refactor to separate these concerns.
+                            // TODO: write a test for this case.
+                            if (with_parts === true || err.code !== 'NoSuchUpload') {
+                                throw err;
+                            }
+                            return this.client.headObject({ Bucket: this.bucket_name, Key: file_id })
+                                .promise()
+                                .then(() => {
+                                    return resolve({
```

Comment:
ouch sorry @Murderlon. I did not see that this was not submitted. I am not familiar with the github workflows yet.

Reply:
Thanks for the effort, you're right. This code could use a refactor but for now we can indeed pass an extra `isFinished` in the `resolve` and return it again right away in the other `then`. 

We should also add this again for the TODO warnings: 
https://github.com/tus/tus-node-server/blob/082e8b91d15ed3f86a8db8267f9e002a8fece60c/.eslintrc#L24

Reply:
I can make those changes.

Reply:
The changes are in @Murderlon 
I had to change the returns in the `then` block, otherwise eslint wouldn't like it (consistent returns). Shouldn't be an issue though since `resolve()` returns `void` anyway.

Reply:
I'll do a round of testing now, just to ensure nothing is broken. I don't like writing this without any type checks or tests...

Reply:
I know, next up for me is setting up a test suite for S3 and rewriting to TS

Reply:
awesome